controllers/deployment.go

```diff
@@ -713,6 +713,36 @@ func extractBaseImageName(imageRef string) (string, error) {
 	return name, nil
 }
 
+// getImagePullPolicy returns the image pull policy for the rollouts container.
+// Order of priority:
+// 1) cr.spec.imagePullPolicy
+// 2) IMAGE_PULL_POLICY environment variable
+// 3) default: corev1.PullIfNotPresent
+func getImagePullPolicy(cr rolloutsmanagerv1alpha1.RolloutManager) corev1.PullPolicy {
+	// First priority: use value from CR spec
+	if cr.Spec.ImagePullPolicy != "" {
+		return cr.Spec.ImagePullPolicy
+	}
+
+	// Second priority: check environment variable
+	envVal := os.Getenv(ImagePullPolicy)
+	if envVal != "" {
+		switch envVal {
+		case string(corev1.PullAlways):
+			return corev1.PullAlways
+		case string(corev1.PullIfNotPresent):
+			return corev1.PullIfNotPresent
+		case string(corev1.PullNever):
+			return corev1.PullNever
+		default:
+			log.Info(fmt.Sprintf("Invalid IMAGE_PULL_POLICY environment variable value: %s, using default: IfNotPresent", envVal))
```

Comment:
IMHO this should return an error, rather than logging at Info level and using default

Reply:
If we return an error, it should fail the reconcile (cause the error to be set on .status.conditions field, thus we don't need a default value)